include/cuco/detail/static_map.inl

```diff
@@ -422,6 +422,83 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
   }
 }
 
+template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
+template <typename Hash, typename KeyEqual>
+__device__
+  thrust::pair<typename static_map<Key, Value, Scope, Allocator>::device_mutable_view::iterator,
```

Comment:
We may want a more descriptive strong type here, e.g., `cuco::insert_result`. A plain `thrust::pair`could mean anything.

Reply:
tbh, I'd love to return a `thrust::optional<iterator>` here.

Reply:
I guess @sleeepyjack's idea is to create a `cuco::insert_result` struct with meaningful member names to explicitly denote the pair-like return type. Something like:
```cpp
struct fetch_insert_result {
  Iterator iter;
  bool is_successful;
};
```
I would preder `thrust::pair<iterator, bool>` since it's std-canonical (see [here](https://en.cppreference.com/w/cpp/container/unordered_map/insert) and [here](https://en.cppreference.com/w/cpp/container/unordered_set/insert)).

Reply:
Yes, this is the idea behind returning a `pair`, that the signature conforms to `std::unordered_map` in the STL.

Reply:
Yes, I find this design decision in the standard questionable from the outset. However, following the standard should minimize confusion. So let's do it this way.